packages/contracts/token_contract/tests/functions/deposit.rs

```diff
@@ -0,0 +1,78 @@
+use crate::utils::{abi_calls::*, test_helpers::setup};
+use fuels::{prelude::*, types::*};
+
+mod success {
+    use super::*;
+
+    #[tokio::test]
+    async fn can_deposit_to_contract() {
+        let meta = setup(None).await;
+        let token_contract = meta.token_contract;
+        let user = meta.user;
+        let params = meta.params;
+        let provider = meta.provider;
+
+        constructor(&token_contract, &params).await;
+
+        let amount = 100;
+        let id = Identity::Address(user.address().into());
+        let deposit = deposit(&token_contract, user, amount).await;
+        let curr_deposit = deposit_of(&token_contract, id).await;
+        assert_eq!(deposit.value, curr_deposit.value);
+
+        let balances = provider
+            .get_contract_balances(&meta.contract_id)
+            .await
+            .unwrap();
+
+        let asset_id_key = format!("{BASE_ASSET_ID:#x}");
```

Comment:
I would try to hoist as many variables to the top of the function as much as reasonable. It makes sense to group things closer to their usage but at the same time that introduces overhead for reading what the test is doing vs where variables are declared.